lib/js/components/OverlayHeader/OverlayHeader.vue

```diff
@@ -0,0 +1,318 @@
+				<template #reference>
+					<icon-button
+						:icon="ICONS.FA_ELLIPSIS_VERTICAL"
+						:size="ICON_BUTTON_SIZES.MEDIUM"
+						:color="ICON_BUTTON_COLORS.NEUTRAL"
+					/>
+				</template>
+				<template #default="{ close }">
+					<slot name="dropdown" :close="close" />
+				</template>
+			</ds-dropdown>
+			<ds-divider
+				class="ds-overlayHeader__divider"
+				:prominence="DIVIDER_PROMINENCES.STRONG"
+				is-vertical
+			/>
+		</template>
+		<icon-button
+			:icon="ICONS.FA_XMARK"
+			:size="ICON_BUTTON_SIZES.MEDIUM"
+			:color="ICON_BUTTON_COLORS.NEUTRAL"
+			@click.native="$emit('close')"
+		/>
+	</div>
+</template>
+
+<style scoped lang="scss">
+@import '../../../styles/settings/spacings';
+@import '../../../styles/settings/media-queries';
+@import '../../../styles/settings/colors/tokens';
+@import '../../../styles/settings/typography/tokens';
+
+$scrollable-mask-width: 20px;
+
+.ds-overlayHeader {
+	align-items: center;
+	background: $color-neutral-background;
+	border-bottom: 2px solid $color-neutral-border-ghost;
+	display: flex;
+	padding: $space-xxs $space-xxxs $space-xxs 0;
+
+	@media #{breakpoint-s()} {
+		padding: $space-xs $space-xxs $space-xs 0;
+	}
+
+	&__accessory {
+		align-self: stretch;
+		margin-left: $space-xxs;
+
+		@media #{breakpoint-s()} {
+			margin-left: $space-xs;
+		}
+	}
+
+	&__content {
+		flex-grow: 1;
+		margin: 0 $space-xxxxs 0 $space-s;
+		// We need to set min-width to allow children to apply ellipsis
+		min-width: 0;
+
+		@media #{breakpoint-s()} {
+			margin-right: $space-xxs;
+		}
+	}
+
+	&__eyebrow {
+		align-items: center;
+		display: flex;
+		margin-bottom: $space-xxxs;
+
+		@media #{breakpoint-s()} {
+			margin-bottom: $space-xxs;
+		}
+	}
+
+	&__eyebrowText {
+		@include info-s-extensive-bold-uppercase;
+
+		color: $color-neutral-text;
+		overflow: hidden;
+		text-overflow: ellipsis;
+		white-space: nowrap;
+		margin-right: $space-xxs;
+	}
+
+	&__eyebrowAccessory {
+		// We need to set min-width to allow children to apply ellipsis
+		min-width: 0;
+	}
+
+	&__main {
+		align-items: center;
+		display: flex;
+	}
+
+	&__titleLeading {
+		display: flex;
+		margin-right: $space-xxs;
+	}
+
+	&__titleWrapper {
+		flex-shrink: 1;
+		overflow: hidden;
+
+		&.-interactive {
+			cursor: pointer;
+		}
+	}
+
+	&__title {
+		@include heading-m-default-bold;
+
+		color: $color-neutral-text-heavy;
+		overflow: hidden;
+		text-overflow: ellipsis;
+		white-space: nowrap;
+
+		&.-desktop {
+			display: none;
+
+			@media #{breakpoint-s()} {
+				display: block;
+			}
+		}
+
+		&.-mobile {
+			@media #{breakpoint-s()} {
+				display: none;
+			}
+		}
+	}
+
+	&__titleTrailingWrapper {
+		display: flex;
+		// flex-shrink: 2 gives some more space for title
+		flex-shrink: 2;
+		margin-left: $space-xxs;
+		overflow-x: auto;
+
+		&::after {
+			align-self: stretch;
+			background: linear-gradient(
+				90deg,
+				rgba(var(--raw-gray-50-rgb), 0) 0%,
+				$color-neutral-background 84.26%
+			);
```

Comment:
It's copy pasted from https://github.com/bethinkpl/wnl-platform/blob/f7b949aee683ca593df6b6159409f3bd6f4cfcf0/resources/assets/js/modules/assessments/components/headers/AssessmentHeader.vue#L357-L361 so maybe it's time to extract it as a DS variable or a mixin?